canonical-web-and-design/maas-ui
# feat(ui): add bond form

## Done

- Add a form for creating bonds.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Create three physical interfaces on the same VLAN.
- Tick two of them and click "Create bond".
- Click "Edit bond members" and untick the interface that is currently marked as primary. The radio button should change to the other interface that is selected.
- Tick a second interface and click "Update bond members".
- Change the bond mode to "802.3ad". The hash policy and LACP rate fields should appear.
- Change the hash policy to "layer3+4". A warning should appear.
- Change the link monitoring field to "MII link monitoring". The monitoring frequency, updelay and downdelay fields should appear.
- Fill out any fields you wish to and submit the form. The bond should get created.

## Fixes

Fixes: #2152.
